src/main.rs

```diff
@@ -209,75 +204,64 @@ fn main() -> Result<(), Box<dyn std::error::Error>> {
     }
 
     if args.cdawg {
-        return match args.disk_path.clone() {
-            Some(path) => {
-                if args.ram {
-                    println!("Building CDAWG in RAM but saving on disk...");
-                    type Mb = RamBacking<N, (DefaultIx, DefaultIx), DefaultIx>;
-                    let mb = Mb::default();
-                    return Ok(build_cdawg::<Mb>(args, mb)?);
-                }
-                println!("Building CDAWG on disk...");
-                type Mb = DiskBacking<N, (DefaultIx, DefaultIx), DefaultIx>;
-                let mb = Mb::new(path);
-                Ok(build_cdawg::<Mb>(args, mb)?)
-            }
-            None => {
-                println!("Building CDAWG in RAM...");
+        return if !args.save_path.is_empty() {
+            if args.ram {
+                println!("Building CDAWG in RAM but saving on disk...");
                 type Mb = RamBacking<N, (DefaultIx, DefaultIx), DefaultIx>;
                 let mb = Mb::default();
-                Ok(build_cdawg::<Mb>(args, mb)?)
+                return Ok(build_cdawg::<Mb>(args, mb)?);
             }
+            println!("Building CDAWG on disk...");
+            type Mb = DiskBacking<N, (DefaultIx, DefaultIx), DefaultIx>;
+            let mb = Mb::new(args.save_path.clone());
+            Ok(build_cdawg::<Mb>(args, mb)?)
+        } else {
+            println!("Building CDAWG in RAM...");
+            type Mb = RamBacking<N, (DefaultIx, DefaultIx), DefaultIx>;
+            let mb = Mb::default();
+            Ok(build_cdawg::<Mb>(args, mb)?)
         };
     }
 
-    match args.disk_path.clone() {
-        Some(path) => println!("DAWG on disk: {}", path),
-        None => println!("DAWG in RAM"),
-    };
+    if !args.save_path.is_empty() {
+        println!("DAWG on disk: {}", args.save_path);
+    } else {
+        println!("DAWG in RAM");
+    }
 
     // Messy, but it works.
     if args.utype == "u16" {
         type E = u16;
-        match args.disk_path.clone() {
-            Some(path) => {
-                type Mb = DiskBacking<N, E, DefaultIx>;
-                let mb = Mb::new(path);
-                run_rusty_dawg::<E, Mb>(args, mb)
-            }
-            None => {
-                type Mb = RamBacking<N, E, DefaultIx>;
-                let mb = Mb::default();
-                run_rusty_dawg::<E, Mb>(args, mb)
-            }
+        if !args.save_path.is_empty() {
```

Comment:
Again, this condition is wrong now, should probably use `--ram` or some other flag